### PR TITLE
File property feature

### PR DESCRIPTION
**Pull request by VirajHasitha**
This feature will use the org.apache.synapse.commons.resolvers.ResolverFactory whenever FilePropertyResolver instantiate.

### CODE REVIEW

**Comment by milindaperera** [comment]
Let's create doc issue as well and add documentation details as there with samples
Also add related git issue to the commit message and PR description

**Comment by VirajHasitha** [comment]
This PR is related to the git issue https://github.com/wso2/product-ei/issues/4799

**Comment by VirajHasitha** [comment]
This PR should merge after the following PRs
https://github.com/wso2/wso2-synapse/pull/1458
https://github.com/wso2/carbon-kernel/pull/2417

File: distribution/src/conf/file.properties [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+stockQuoteEP=http://localhost:9000/services/SimpleStockQuoteService
```
Comment (milindaperera) [inline distribution/src/conf/file.properties]: Let's remove this stockQuoteEP and move it to the test case configuration files.
And add comment about the default file specifying what we should put in this file

**Comment by milindaperera** [comment]
merge after Synapse v2.1.7-wso2v140 upgrade

File: distribution/src/conf/file.properties (hunk 1 above)
Comment (VirajHasitha) [reply distribution/src/conf/file.properties]: Fixed with the commit https://github.com/wso2/product-ei/pull/4812/commits/d5b6c5c11a99aebb97d38259d0d74819b1607993

**Comment by BeParis** [comment]
Hi devs,

I'm testing out of the box API manager 3.0.0 on my Mac, then try to edit a new Service Provider I just created, following instructions at
https://is.docs.wso2.com/en/5.9.0/learn/configuring-saml2-web-single-sign-on/


[cid:36C39DB0-ABF1-4D7C-AC27-7C6F38276B6D]


When I click "Configure" SAML2 Web SSO Configuration the main section page is empty;
In the logs:

[2019-12-05 10:14:01,492] ERROR - RenderTagSupport Error during rendering
javax.servlet.jsp.JspException: IO Error executing tag: JSPException while including path '/sso-saml/add_service_provider.jsp'. ServletException while including page.

Same behavior if I try to edit an existing service provider such as "admin_DefaultApplication_PRODUCTION"


Thanks for your help,
Bernard

**Pull request merged**